src/take.test.ts

```diff
@@ -16,3 +24,139 @@ describe("data_last", () => {
     expect(pipe([1, 2, 3, 4, 3, 2, 1], take(0))).toEqual([]);
   });
 });
+
+describe("typings", () => {
+  it("handles strict tuples", () => {
+    const strict = ["1", 2, true] as [string, number, boolean];
+    expectTypeOf(take(strict, 2)).toEqualTypeOf<[string, number]>();
+
+    const strictReadonly = ["1", 2, true] as readonly [string, number, boolean];
+    expectTypeOf(take(strictReadonly, 2)).toEqualTypeOf<[string, number]>();
+  });
+
+  it("handles literal unions", () => {
+    const unions = ["cat", "mouse"] as ["cat" | "dog", "mouse" | "chicken"];
+    expectTypeOf(take(unions, 1)).toEqualTypeOf<["cat" | "dog"]>();
+
+    const unionsReadonly = ["cat", "mouse"] as readonly [
+      "cat" | "dog",
+      "mouse" | "chicken",
+    ];
+    expectTypeOf(take(unionsReadonly, 1)).toEqualTypeOf<["cat" | "dog"]>();
+  });
+
+  it("handles rest tails", () => {
+    const restTail = ["1", 2, 3] as [string, ...Array<number>];
+    expectTypeOf(take(restTail, 1)).toEqualTypeOf<[string]>();
+    expectTypeOf(take(restTail, 2)).toEqualTypeOf<[string, ...Array<number>]>();
+
+    const restTailReadonly = ["1", 2, 3] as readonly [string, ...Array<number>];
+    expectTypeOf(take(restTailReadonly, 1)).toEqualTypeOf<[string]>();
+    expectTypeOf(take(restTailReadonly, 2)).toEqualTypeOf<
+      [string, ...Array<number>]
+    >();
+  });
+
+  it("handles rest heads", () => {
+    const restHead = [1, 2, 3, "foo"] as [...Array<number>, string];
+    expectTypeOf(take(restHead, 3)).toEqualTypeOf<[...Array<number>, string]>();
```

Comment:
similarly this could be `[string] | [number, string] | [number, number, string] | [number, number, number]`